xwiki-platform-core/xwiki-platform-image/xwiki-platform-image-lightbox/xwiki-platform-image-lightbox-test/xwiki-platform-image-lightbox-test-docker/src/test/it/org/xwiki/image/lightbox/test/ui/LightboxIT.java

```diff
@@ -0,0 +1,445 @@
+        testUtils.getDriver().navigate().refresh();
+
+        Optional<Lightbox> lightboxOptional = lightboxPage.openLightboxAtImage(0);
+        assertTrue(lightboxOptional.isPresent());
+
+        if (lightboxOptional.isPresent()) {
+            Lightbox lightbox = lightboxOptional.get();
+            assertTrue(lightbox.isDisplayed());
+
+            // Start auto play.
+            lightbox.toggleSlideshow();
+            assertTrue(lightbox.waitForSlide(1));
+            assertTrue(lightbox.waitForSlide(2));
+            assertTrue(lightbox.waitForSlide(0));
+
+            // Stop auto play.
+            lightbox.toggleSlideshow();
+            assertFalse(lightbox.waitForSlide(1));
+        }
+    }
+
+    @Test
+    @Order(9)
+    void openMissingImage(TestUtils testUtils, TestReference testReference, TestConfiguration testConfiguration)
+    {
+        testUtils.updateObject(LIGHTBOX_CONFIGURATION_REFERENCE, LIGHTBOX_CONFIGURATION_CLASSNAME, 0,
+            "isLightboxEnabled", "1");
+
+        testUtils.createPage(testReference, this.getSimpleImage(images.get(2)));
+        lightboxPage = new LightboxPage();
+
+        // Make sure that the images are displayed.
+        testUtils.getDriver().navigate().refresh();
+
+        Optional<Lightbox> lightboxOptional = lightboxPage.openLightboxAtImage(0);
+        assertTrue(lightboxOptional.isPresent());
+
+        if (lightboxOptional.isPresent()) {
+            Lightbox lightbox = lightboxOptional.get();
+            assertTrue(lightbox.isDisplayed());
+
+            assertTrue(lightbox.isImageMissing());
+            assertEquals("0", lightbox.getSlideIndex());
+            assertEquals(images.get(2), lightbox.getCaption());
+            assertEquals("", lightbox.getTitle());
+            assertEquals("", lightbox.getDate());
+        }
+    }
+
+    @Test
+    @Order(10)
+    void openFullscreen(TestUtils testUtils, TestReference testReference, TestConfiguration testConfiguration)
+    {
+        testUtils.updateObject(LIGHTBOX_CONFIGURATION_REFERENCE, LIGHTBOX_CONFIGURATION_CLASSNAME, 0,
+            "isLightboxEnabled", "1");
+
+        testUtils.createPage(testReference, this.getSimpleImage(images.get(0)));
+        lightboxPage = new LightboxPage();
+
+        lightboxPage.attachFile(testConfiguration.getBrowser().getTestResourcesPath(), images.get(0));
+
+        // Make sure that the images are displayed.
+        testUtils.getDriver().navigate().refresh();
+
+        Optional<Lightbox> lightboxOptional = lightboxPage.openLightboxAtImage(0);
+        assertTrue(lightboxOptional.isPresent());
+
+        if (lightboxOptional.isPresent()) {
+            Lightbox lightbox = lightboxOptional.get();
+            assertTrue(lightbox.isDisplayed());
+
+            Optional<WebElement> fullscreenOn = lightbox.toggleFullscreen();
+            if (fullscreenOn.isPresent()) {
+                assertTrue(fullscreenOn.get().isDisplayed());
+            }
```

Comment:
I don't think the test should check if the fullscreen element is displayed. If the ``toggleFullscreen()`` didn't throw an exception then I assume the switch was successful (i.e. it's the page object that should ensure that the fullscreen mode is ready), and I can use the return value (a boolean is better for this) to check the state (fullscreen on or off).